dbt/macros/not_all_null.sql

```diff
@@ -1,9 +1,12 @@
-{% test not_all_null(model, column_name) %}
+{% test not_all_null(model, column_name, filter=None) %}
```

Comment:
Hmm. Will this work for `etl-fast` cases where we're limited to very recent dates, such that there are no rows in the table before the deprecation date?

Given a table with

* `report_date` starting at 2023-01-01 and going forward from there
* `deprecated_column` all null, bc it stopped being published in 2020

and a schema.yml with

```
columns:
  - name: deprecated_column
    data_tests:
      - not_all_null:
          filter: "WHERE report_date < '2020-01-01'"
```

Will this pass or fail?

Reply:
Schema tests fail only when there are rows returned, so if a filter results in zero rows being returned, that would mean the test passes.